PYIC-1295: Validate KBV check

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed
Validate KBV check

### Why did it change

This adds logic to the validatecricheck lambda to allow us to determine
if a KBV check has reached a high enough confidence level to proceed
with an M1A journey.

I've gone for a different approach for how to organise the validators.
I've attempted to use composition as a technique to allow encapsulation
of the policies for different credential issuers, whilst allowing common
logic to be shared. Does this make things unnecessarily complicated or
easy to understand? You tell me...

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-1295](https://govukverify.atlassian.net/browse/PYIC-1295)

Wynndow committed:
PYIC-1295: Validate KBV check

This adds logic to the validatecricheck lambda to allow us to determine
if a KBV check has reached a high enough confidence level to proceed
with an M1A journey.

I've gone for a different approach for how to organise the validators.
I've attempted to use composition as a technique to allow encapsulation
of the policies for different credential issuers, whilst allowing common
logic to be shared. Does this make things unnecessarily complicated or
easy to understand? You tell me...

dannyhayes-gds commented on lambdas/validatecricheck/src/main/java/uk/gov/di/ipv/core/validatecricheck/validation/CriCheckValidator.java:
```diff
@@ -1,28 +1,24 @@
     public static final String CRI_ID_UK_PASSPORT = "ukPassport";
+    public static final String CRI_ID_KBV = "kbv";
     public static final String EVIDENCE = "evidence";
     public static final int GPG_45_M1A_STRENGTH_SCORE = 4;
```
Don't think you need this static vars for the strength and validity here anymore as you have them in the new PassportEvidenceValidator class

dannyhayes-gds commented on lambdas/validatecricheck/src/main/java/uk/gov/di/ipv/core/validatecricheck/validation/KbvEvidenceValidator.java:
```diff
@@ -0,0 +1,19 @@
+        KbvEvidence evidence =
+                gson.fromJson(evidenceArray.get(ONLY_ELEMENT).toString(), KbvEvidence.class);
+
+        return evidence.getVerificationScore() >= GPG_45_M1A_VERIFICATION_SCORE;
```
Do we need to check for contra indicators here as well? Not sure if thats a thing for kbv?

Wynndow replied:
No - KBV will never return contra-indicators. I checked this with @idrop yesterday and he confirmed. You can see it [the RFC that they don't exist](https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0024-identity-vc-for-checks.md#knowledge-based-verification-check).

Wynndow committed:
PYIC-1295: Use score constants defined on passport class

The scores for a passport check were defined twice. This was a hangover
from a previous version. This switches to those defined on the passport
evidence class.
